### PR TITLE
upgrade hugo, swap to hugo modules, upgrade docsy

### PR DESCRIPTION
**Pull request by jeefy**
- Upgrades Hugo to 0.120.3
- Upgrades Docsy to 0.7.2
- Swaps to using `hugo mod` 
- Dockerfile workaround for https://github.com/gohugoio/hugo/issues/10839
- Updates `hack/gen-content.sh` to suppress `grep` warnings (see https://github.com/koalaman/shellcheck/issues/2573 for example)
- Content pulled in from a `gen-content` run.

### CODE REVIEW

**Comment by jeefy** [comment]
The preview has some styling errors, presumably from the Docsy update. I can start trying to chase them down but any help there would be appreciated. 🙃 

/hold

**Comment by jeefy** [comment]
Pretty sure I fixed the CSS issues post-upgrade

/hold cancel

/assign @mrbobbytables

File: content/en/code-of-conduct.md
Comment (mrbobbytables) [inline content/en/code-of-conduct.md:1]: Was this addition accidental?^^;;

File: hack/gen-content.sh
Comment (mrbobbytables) [inline hack/gen-content.sh:1]: The grep command changes broke link detection. =/ Even small changes to these tend to break links in some way. 

**Example:**
https://deploy-preview-456--kubernetes-contributor.netlify.app/docs/guide/#community-expectations-and-roles
 - "Community Expectations" generates a link with: `/docs/guide/expectations.md` when it should be `/docs/guide/expectations` (does not strip trailing md and link 404's)
    
It's the case for a good chunk of the links across the pages. I looked at various others in the guide section and its like that.

**Review by mrbobbytables — COMMENTED** [review]
Sadly theres still some broken stuff =/
Here is a picture of the current site:
current


Here is the preview of this PR:
- titlebar is in the wrong position
- search is missing
- Text on the page is bigger, but I don't think its an issue; just a difference vs the old one.
new

**Review by dipesh-rawat — COMMENTED** [review]
There are few more issues in addition to the concerns highlighted in review comment https://github.com/kubernetes/contributor-site/pull/456#pullrequestreview-1738730383 regarding the main navigation bar and the absence of a search feature. Specifically, the mobile view for smaller screens, unfortunately has some issues such as the navigation bar size seems to be cutting off text and the hamburger menu for navigating between pages is non-functional.

[Deploy Preview](https://deploy-preview-456--kubernetes-contributor.netlify.app/docs/)
_(Attached the screenshot below for reference)_

mobile-view

**Review by sftim — COMMENTED** [review]
I recommend splitting the version bump for Hugo into its own commit.

**Comment by sftim** [comment]
The links in https://www.kubernetes.dev/docs/contributor-cheatsheet/ to other languages work, but the links in https://deploy-preview-456--kubernetes-contributor.netlify.app/docs/contributor-cheatsheet/ do not.

Can we fix that before this gets LGTM?

**Comment by sftim** [comment]
@jeefy what would you like to happen about https://github.com/kubernetes/contributor-site/pull/456#issuecomment-1876086658? I can try to rally some help there (if you'd like it).

**Pull request closed without merging**